pkg/signature/manifest.go

```diff
@@ -0,0 +1,162 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package signature
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"text/template"
+
+	"github.com/itchyny/gojq"
+	"sigs.k8s.io/yaml"
+
+	"github.com/aws/eks-anywhere/pkg/constants"
+	anywherev1alpha1 "github.com/aws/eks-anywhere/release/api/v1alpha1"
+)
+
+// GojqTemplate is used to build a gojq filter expression that deletes the desired fields.
+var GojqTemplate = template.Must(template.New("gojq_query").Funcs(
+	template.FuncMap{
+		"StringsJoin": strings.Join,
+		"Escape": func(in string) string {
+			// We need to escape '.' for certain gojq path usage
+			// to avoid ambiguities in the path expressions.
+			return strings.ReplaceAll(in, ".", "\\\\.")
+		},
+	},
+).Parse(`
+del({{ StringsJoin .Excludes ", " }})
+`))
+
+// ValidateSignature validates the signature annotation of the bundles object using KMS public key.
+func ValidateSignature(bundle *anywherev1alpha1.Bundles, pubKey string) (valid bool, digest [32]byte, yml []byte, err error) {
+	bundleSig := bundle.Annotations[constants.SignatureAnnotation]
+	if bundleSig == "" {
+		return false, [32]byte{}, yml, fmt.Errorf("missing signature annotation")
+	}
+
+	digest, yml, err = getDigest(bundle)
+	if err != nil {
+		return false, [32]byte{}, yml, err
+	}
+
+	sig, err := base64.StdEncoding.DecodeString(bundleSig)
+	if err != nil {
+		return false, digest, yml, fmt.Errorf("signature in metadata isn't base64 encoded")
+	}
+
+	pubdecoded, err := base64.StdEncoding.DecodeString(pubKey)
+	if err != nil {
+		return false, digest, yml, fmt.Errorf("unable to decode the public key (not base 64)")
+	}
+
+	pubparsed, err := x509.ParsePKIXPublicKey(pubdecoded)
+	if err != nil {
+		return false, digest, yml, fmt.Errorf("unable parse the public key (not PKIX): %v", err)
+	}
+
+	pubkey := pubparsed.(*ecdsa.PublicKey)
+
+	return ecdsa.VerifyASN1(pubkey, digest[:], sig), digest, yml, nil
+}
+
+// getDigest converts the Bundles manifest to JSON, excludes certain fields, then
+// computes the SHA256 hash of the filtered manifest. It returns the digest and
+// the final bytes used to produce that digest.
+func getDigest(bundle *anywherev1alpha1.Bundles) ([32]byte, []byte, error) {
+	var zero [32]byte
+
+	// Marshal Bundles object to YAML
+	yamlBytes, err := yaml.Marshal(bundle)
+	if err != nil {
+		return zero, nil, fmt.Errorf("marshalling bundle to YAML: %v", err)
```

Comment:
might think about this: %v -> %w.

"Wrap an error to expose it to callers. Do not wrap an error when doing so would expose implementation details."
reference: https://go.dev/blog/go1.13-errors#wrapping-errors-with-w